flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1183,8 +1183,12 @@ public void notifyCheckpointComplete(final long checkpointID) {
 				@Override
 				public void run() {
 					try {
-						invokable.notifyCheckpointComplete(checkpointID);
-					} catch (Throwable t) {
+						invokable.notifyCheckpointCompleteAsync(checkpointID);
```

Comment:
Is the future used at all? Do we need a #get here?

Reply:
Correct me if I'm wrong, but I think, we don't need to wait for a result here.

Reply:
if we do not wait, then we will miss the error handling? 

But please first respond to my comment about double/nested mailbox/enqueuing pattern.